provider/cluster/inventory.go

```diff
@@ -201,7 +209,17 @@ loop:
 						continue
 					}
 
+					allocatedPrev := res.allocated
 					res.allocated = ev.Status == event.ClusterDeploymentDeployed
+					allocatedChanged := res.allocated != allocatedPrev
+					if allocatedChanged {
```

Comment:
`if res.allocated != allocatedPrev {`

Reply:
I agree we could eliminate the local variable here. I usually like to do this however because the comparison `!=` is just a comparison. But by assigning it to a local variable with a name, you can document that meaning of the comparison.

Reply:
I think pattern `curr != prev` is self descriptive though